Components/LearningCore/Source/gov/sandia/cognition/learning/algorithm/clustering/OptimizedAffinityPropagation.java

```diff
@@ -0,0 +1,261 @@
+package gov.sandia.cognition.learning.algorithm.clustering;
+
+import gov.sandia.cognition.learning.algorithm.clustering.AffinityPropagation;
+import gov.sandia.cognition.math.DivergenceFunction;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.stream.DoubleStream;
+
+public class OptimizedAffinityPropagation<T>

```

Comment:
Please add a class-level Javadoc with @author tag added.  I think you could take much of the text from the AffinityPropagation's class-level Javadoc with only minimal additions to describe how this is optimized versus that.  While I realize that Git preserves who submits changes there are times that the code was written by another and submitted by an organization leader, so we do like to have authors explicitly called out with the code.

Reply:
Done. I have added Javadoc with my name as author. I don't know if it's right that the license part of the doc is still there after my commit. I wrote "Marco Pezzulla" also as author of the constructors. Also here, i'm not sure it's right.